src/Breadcrumbs.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Bulma;
+
+use Yiisoft\Arrays\ArrayHelper;
+use Yiisoft\Html\Html;
+use Yiisoft\Widget\Exception\InvalidConfigException;
+
+/**
+ * The Bulma breadcrumb is a simple navigation component that only requires
+ *
+ * For example,
+ *
+ * ```php
+ * echo Breadcrumbs::widget()
+ *     ->links(['label' => !empty($this->params['breadcrumbs']) ? $this->params['breadcrumbs'] : []]);
+ * ```
+ */
+class Breadcrumbs extends Widget
+{
+    private bool $encodeLabels = true;
+    private array $homeLink = [];
+    private string $itemTemplate = "<li>{link}</li>\n";
+    private string $itemTemplateActive = "<li class=\"is-active\"><a aria-current=\"page\">{label}</li>\n";
+    private array $links = [];
+    private array $options = [];
+    private array $optionsItems = [];
+
+    protected function run(): string
```

Comment:
I suggest dividing it into methods for readability:

```php
    protected function run(): string
    {
        if (empty($this->links)) {
            return '';
        }

        $this->buildOptions();

        return Html::tag('nav', Html::tag('ul', implode('', $this->buildLinks()), $this->optionsItems), $this->options);
    }

    private function buildOptions(): void
    {
        if (!isset($this->options['id'])) {
            $this->options['id'] = $this->createId();
        }

        $this->options = $this->addOptions($this->options, 'breadcrumb');
        $this->options = array_merge(['aria-label' => 'breadcrumbs'], $this->options);
    }

    private function createId(): string
    {
        return "{$this->getId()}-breadcrumbs";
    }

    private function buildLinks(): array
    {
        $links = [$this->getHomeLink()];
        foreach ($this->links as $link) {
            $links[] = $this->createLink($link);
        }

        return $links;
    }

    private function getHomeLink(): string
    {
        if ($this->homeLink === array()) {
            return $this->createHomeLink();
        }

        return $this->renderItem($this->homeLink, $this->itemTemplate);
    }

    private function createHomeLink(): string
    {
        return $this->renderItem(
            [
                'label' => 'Home',
                'url' => '/',
            ],
            $this->itemTemplate
        );
    }

    private function createLink($link): string
    {
        if (!\is_array($link)) {
            $link = ['label' => $link];
        }

        return $this->renderItem($link, isset($link['url']) ? $this->itemTemplate : $this->itemTemplateActive);
    }

```

Reply:
implement `buildOptions()`, done.